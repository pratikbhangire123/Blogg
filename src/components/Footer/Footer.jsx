function Footer() {
  return (
    <footer className="w-full p-2 text-center text-xs md:text-sm font-thin text-gray-50 bg-gray-900">
      &copy; Copyright 2024. All Rights Reserved by&nbsp;
      <span className="font-normal">Blogg</span>.
    </footer>
  );
}

export default Footer;
