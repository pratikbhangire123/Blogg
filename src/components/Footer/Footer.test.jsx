import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Footer from "./Footer";


it("should render a copyright statement & logo", () => {
  render(<Footer />);

  const footer = screen.getByTestId("footer");

  expect(footer).toBeInTheDocument();

  expect(footer).toHaveTextContent(
    /^Â© Copyright 2024. All Rights Reserved by Blogg.$/
  );
});

it("should matches snapshot", () => {
  const footerComponent = renderer.create(<Footer />).toJSON();

  expect(footerComponent).toMatchSnapshot();
});
