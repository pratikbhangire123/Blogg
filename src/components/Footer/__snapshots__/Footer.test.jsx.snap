// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should matches snapshot 1`] = `
<footer
  className="w-full p-2 text-center text-xs md:text-sm font-thin text-gray-50 bg-gray-900"
  data-testid="footer"
>
  © Copyright 2024. All Rights Reserved by 
  <span
    className="font-normal"
  >
    Blogg
  </span>
  .
</footer>
`;
